### ISSUE TITLE
TFMA/Evaluator fails when training Keras model with MultiWorkerMirroredStrategy

### ISSUE THREAD

**Issue by casassg** [tensorflow/tfx#2894]
Getting an issue when training my model in AI Platform Training in `MultiWorkerMirroredStrategy`. When I try to run Evaluator on the resulting model, I get an issue from tensorflow not knowing how to assign a correct device: 

```
RuntimeError: tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot assign a device for operation StatefulPartitionedCall/functional_1/deep_and_cross_network/dnn_batch_norm_01/Cast/allreduce/CollectiveReduce: {{node StatefulPartitionedCall/functional_1/deep_and_cross_network/dnn_batch_norm_01/Cast/allreduce/CollectiveReduce}} was explicitly assigned to /job:worker/replica:0/task:1/device:CPU:0 but available devices are [ /job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:XLA_CPU:0 ]. Make sure the device specification refers to a valid device.
```

Seems tensorflow model saves the assignation for sine tasks to the workers devices, however this fails in Dataflow as it can't detect those specific devices.

**Comment by rmothukuru** [comment]
@casassg,
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by casassg** [comment]
https://colab.research.google.com/drive/1Iv8RoUSZCjjECooQ-vDnKSNoptPxxv6w?usp=sharing

It looks like this but launching to AI Platform and multiple workers. Not sure how to launch w multiple workers in local setup, but please do let me know if there's a way to test it.

**Comment by 1025KB** [comment]
The link you send is not on cloud, it's just colab (similar to local), multiworker strategy won't work.

This is an [example](https://github.com/tensorflow/tfx/blob/master/tfx/examples/chicago_taxi_pipeline/taxi_pipeline_kubeflow_gcp.py) running on AI platform

**Comment by casassg** [comment]
Yup, I know. However, I dont have a way to run on AI Platform due to how VPC is setup for us (no access to AI Platform from Collab). I can setup a GCP personal project under my personal account, but that is quite a lot of effort I would rather avoid if possible.

**Comment by 1025KB** [comment]
For multiworker strategy to work, you need to setup the [training args](https://github.com/tensorflow/tfx/blob/master/tfx/examples/penguin/penguin_pipeline_kubeflow_gcp.py#L228) thus Cloud will start multiple workers to run the code with distributed strategy 

Here is a general [instruction](https://cloud.google.com/ai-platform/training/docs/distributed-training-details) from cloud side

**Comment by arghyaganguly** [comment]
@casassg , please confirm if this can be closed.
